docs/developer/explanation/debian-package-publishing.rst

```diff
@@ -0,0 +1,80 @@
+Debian package publishing
+=========================
+
+The Debian package lifecycle describes the process from a `source package upload <https://documentation.ubuntu.com/project/how-ubuntu-is-made/concepts/package-format/#source-packages>`_ to the `Ubuntu package archive <https://documentation.ubuntu.com/project/how-ubuntu-is-made/concepts/package-archive/>`_.
+This involves multiple steps in which the original source package is verified, built into a `binary package <https://documentation.ubuntu.com/project/how-ubuntu-is-made/concepts/package-format/#binary-packages>`_ and finally published into the package archive.
+
+Please note that this document is not describing the review processes involved in accepting package changes or the migrations between suites, but rather the technical steps involved in the publishing process.
+
+#####################
+Source package upload
+#####################
+
+The first step of the publishing process is to upload a source package with a signed
+source ``.changes`` file. The signed document references and includes checksums of the files to upload and
+the source package. These files are usually uploaded with ``dput`` or ``dput-ng``, which use
+FTP or SFTP to push the files to Launchpad.
+
+The Ubuntu upload processor (``process-upload.py``) runs every minute. When it finds a newly
+uploaded source package, it performs various checks, including:
+
+- ensuring the signature on the changes file is valid and that the key used is associated with a user who has permission to upload to the target distribution/series/archive
+- ensuring all checksums match
+- confirming that it really is a source upload
+
+..
+    TODO: Create and link to librarian explainer
+
+Once all checks pass, the script uploads the files to the librarian and creates a queue entry (``PackageUpload``) in the database.
+
+If the upload is automatically accepted (e.g. for the devel series), the system immediately creates a ``PENDING`` source package
+publishing history entry and creates all necessary binary package build requests. Otherwise these are created later when the
+package is accepted from the queue.
+
+########
+Building
+########
+
+The Launchpad build farm is controlled by the ``buildd-manager`` daemon, which manages multiple
+``launchpad-buildd`` instances for all supported architectures.
+
+..
+    Explain chroot?
+
+For a binary package build, a ``buildd`` instance is given a list of artifacts to download from
```

Comment:
I think this should be `launchpad-buildd`. As we also have `buildd-manager`, let's try to keep things named the same way.